src/gwt/src/org/rstudio/core/client/command/AppCommand.java

```diff
@@ -200,6 +201,9 @@ public void setVisible(boolean visible)
          visible_ = visible;
          handlers_.fireEvent(new VisibleChangedEvent(this));
       }
+      
+      if (Desktop.isDesktop())
```

Comment:
Similar here?

Reply:
The `AppCommand` is supposed to notify the desktop when its state changes. This happens here:

https://github.com/rstudio/rstudio/blob/master/src/gwt/src/org/rstudio/core/client/command/AppCommand.java#L110-L116

Reply:
Gotcha. I think the behavior I was seeing was related to a satellite window thinking it wasn't a desktop window, which should be alleviated through separate changes in this PR (but I'll test to verify).